security/src/main/java/io/airlift/security/utils/SecurityUtil.java

```diff
@@ -0,0 +1,16 @@
+package io.airlift.security.utils;
+
+import org.apache.shiro.SecurityUtils;
+import org.apache.shiro.subject.Subject;
+
+public class SecurityUtil
+{
+    public static final String REALM_IN_CHALLENGE = "X-Realm-In-Challenge";
+
+    // TODO: convert the Shiro's Subject to javax.security.auth.Subject
+    public static Subject getSubject()
```

Comment:
The code in this PR doesn't depend on Shiro except for this method, so this seems like the wrong place to put this. It belongs in whichever code actually uses Shiro for auth.